lib/sles4sap_publiccloud_basetest.pm

```diff
@@ -28,6 +28,17 @@ sub cleanup {
         # Skip cleanup if ansible inventory is not present (deployment could not have been done without it)
         next if (script_run 'test -f ' . qesap_get_inventory(get_required_var('PUBLIC_CLOUD_PROVIDER')));
 
+        record_info('Cleanup', `Executing peering cleanup (if peering is present)`);
+        my $rg = qesap_az_get_resource_group();
```

Comment:
It has to be in a `if` like:
```
if check_var('PUBLIC_CLOUD_PROVIDER', 'AZURE')
```

Reply:
I propose to add an if here for the moment, I'd suggest to have at least 2  implementation before to think about how to re-factor to a unit function

Reply:
done, also added check for `IS_MAINTENANCE` and pasted new VRs (still running, but the last one completed and it behaves as expected)